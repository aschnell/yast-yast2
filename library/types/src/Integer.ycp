/**
 * File:	modules/Integer.ycp
 * Package:	yast2
 * Summary:	Integer routines
 * Authors:	Arvin Schnell <aschnell@suse.de>
 *
 * $Id: Integer.ycp 45945 2008-04-01 19:41:01Z aschnell $
 */

{
    module "Integer";
    textdomain "base";


    /**
     * Generate a list<integer> with the integers from 0 to n-1.
     */
    global list<integer> Range(integer n)
    {
	list<integer> ret = [];
	integer i = 0;
	while (i < n)
	{
	    ret = add(ret, i);
	    i = i + 1;
	}
	return ret;
    }
}
