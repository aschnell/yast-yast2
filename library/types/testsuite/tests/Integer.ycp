{

include "testsuite.ycp";
import "Integer";

DUMP("Integer::Range");
TEST(``(Integer::Range(0)), [], nil);
TEST(``(Integer::Range(10)), [], nil);

DUMP("Integer::RangeFrom");
TEST(``(Integer::RangeFrom(5, 10)), [], nil);

DUMP("Integer::IsPowerOfTwo");
TEST(``(Integer::IsPowerOfTwo(0)), [], nil);
TEST(``(Integer::IsPowerOfTwo(1)), [], nil);
TEST(``(Integer::IsPowerOfTwo(2)), [], nil);
TEST(``(Integer::IsPowerOfTwo(3)), [], nil);
TEST(``(Integer::IsPowerOfTwo(4)), [], nil);

TEST(``(Integer::IsPowerOfTwo(1024-1)), [], nil);
TEST(``(Integer::IsPowerOfTwo(1024)), [], nil);
TEST(``(Integer::IsPowerOfTwo(1024+1)), [], nil);

TEST(``(Integer::IsPowerOfTwo(1024*1024*1024*1024-1)), [], nil);
TEST(``(Integer::IsPowerOfTwo(1024*1024*1024*1024)), [], nil);
TEST(``(Integer::IsPowerOfTwo(1024*1024*1024*1024+1)), [], nil);

}
