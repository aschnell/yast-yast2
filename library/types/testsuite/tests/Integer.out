Dump	Integer::Range
Return	[]
Return	[0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
Dump	Integer::RangeFrom
Return	[5, 6, 7, 8, 9]
Dump	Integer::IsPowerOfTwo
Return	false
Return	true
Return	true
Return	false
Return	true
Return	false
Return	true
Return	false
Return	false
Return	true
Return	false
Dump	Integer::Sum
Return	0
Return	1
Return	5
